pkg/skaffold/docker/image.go

```diff
@@ -97,6 +106,88 @@ func RunPush(ctx context.Context, cli DockerAPIClient, ref string, out io.Writer
 	return streamDockerMessages(out, rc)
 }
 
+func AddTag(src, target string) error {
+	srcRef, err := resolveReference(src)
+	if err != nil {
+		return errors.Wrap(err, "getting source reference")
+	}
+
+	auth, err := authn.DefaultKeychain.Resolve(srcRef.Context().Registry)
+	if err != nil {
+		return err
+	}
+
+	targetRef, err := resolveReference(target)
+	if err != nil {
+		return errors.Wrap(err, "getting target reference")
+	}
+
+	fmt.Println()
+
+	return addTag(srcRef, targetRef, auth, http.DefaultTransport)
+}
+
+func resolveReference(identifier string) (name.Reference, error) {
+	// try tag first
+	tag, err := name.NewTag(identifier, name.WeakValidation)
```

Comment:
I think there's a new ParseReference method in the client library that does this.